src/__tests__/jest-reporters/utils.js

```diff
@@ -0,0 +1,45 @@
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+const filterFileNamesFromGitStatusOutput = (output) => {
+  const files = output
+    .split('\n')
+    .map((line) => line.trim())
+    // Make sure we ignore deleted files.
+    .filter((line) => !line.startsWith('D'))
+    .map((line) => {
+      // Return the new name of a renamed file.
+      if (line.startsWith('RM')) {
+        return line.substring(line.indexOf('->'));
+      }
+
+      return line;
+    })
+    .map((line) => line.split(' '))
+    // .flatMap()
+    .reduce((chunks, chunk) => chunks.concat(chunk), [])
+    // We assume no filename can be smaller than 3 to filter the short format
+    // statuses: https://git-scm.com/docs/git-status#_short_format.
+    .filter((chunk) => chunk.length > 3)
+    // Allow directories OR JS/JSX files
+    .filter((filename) => /(\/|\.jsx?)$/.test(filename));
```

Comment:
Needs moar TypeScript

```diff
diff --git a/src/__tests__/jest-reporters/utils.js b/src/__tests__/jest-reporters/utils.js
index 587385b..2e56f7d 100644
--- a/src/__tests__/jest-reporters/utils.js
+++ b/src/__tests__/jest-reporters/utils.js
@@ -22,7 +22,7 @@ const filterFileNamesFromGitStatusOutput = (output) => {
     // statuses: https://git-scm.com/docs/git-status#_short_format.
     .filter((chunk) => chunk.length > 3)
     // Allow directories OR JS/JSX files
-    .filter((filename) => /(\/|\.jsx?)$/.test(filename));
+    .filter((filename) => /(\/|\.[tj]sx?)$/.test(filename));
 
   return files;
 };
```

Reply:
I believe we should not even have the `j`, only `tsx?`

Reply:
I don't see the harm in linting `js` files if they exist. For example, this patch adds two `js` files and we probably want them linted, don't we?

Reply:
ah, right.